### PR TITLE
Added custom enumerable 

### PR DESCRIPTION
**Pull request by AbelG101**
## Changes
- Added a list class with 'each' iteration implementation and test cases.
- Added a custom enumerable module with all, any & filter implementation

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+require_relative 'my_enumerable'
+class MyList
+  include MyEnumerable
+
+  def initialize(list)
```
Comment (Ismailco) [inline my_list.rb]: - [ ] MyList class should have the #initialize method that accepts multiple arguments, not just one argument, kindly consider implementing this so your class accepts more than one argument

e.g
```ruby
def initialize(*args)
        @list = []
        args.each { |arg| @list << arg }
    end
```

File: my_list.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @list = list
+  end
+
+  attr_reader :list
+
+  def each
+    puts yield @list if block_given?
+  end
+end
```
Comment (Ismailco) [inline my_list.rb]: - [ ] The #each method should yield successive members of @list, not individual ones, kindly consider modifying your code so `irb> list = MyList.new(1, 2, 3, 4)` returned `#<MyList: @list=[1, 2, 3, 4]>`

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @AbelG101, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
Hi @muskan2532150, 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**